### PR TITLE
9x10 Holodeck and New Holodeck Simulations

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
Enlarges the holodeck areas on all stations to be 9x10 and adds two new simulations: space checkers and skate park. All previous holodeck simulations were changed to be 9x10 without adding anything that would break their balance. Skate park is a bunch of tables to grind on with skateboards, and space checkers is a chess board set up with white/black pawns in the checkers starting position. 

Screenshot_137
New Space Checkers Sim
Screenshot_139
Not Shitty Space Chess
Screenshot_128
New Skate Park Sim 
[heres some gameplay for the skate park, which is slightly outdated](https://imgur.com/a/s1SsGES)
I'm also debating giving holodeck skateboards a smaller instability factor, so that you can do more grinding/ollies before hitting stamina crit. Currently it really doesn't take much before you're stunned
EDIT: Thanks to uomo for fixing both major bugs that made the skate park unusable 

## Why It's Good For The Game
Was originally started because space chess is horrible and boring, so it needed to have space for a full chess board. This PR allows all sims to have about 30 extra tiles of space, and the new sims added are very reliant on said extra space. Space checkers is just as space dependent as space chess and the skate park is still somewhat cramped even with the space. The main thing this PR does is scale everything up, which allows more people to use it at once. More people can fit in the lounge, the courtroom, both thunderdomes, etc. More of the station can rely on the holodeck for competitions: chess matches against the clown, skating matches against the captain for aa, etc. 

## Changelog
:cl:
tweak: All holodeck sims now 9x10,
tweak: All stations can accommodate the enlarged holodecks
add: added skatepark sim
add: added space checkers sim
/:cl:

All the rest of the new sims:

Screenshot_123
basketball court
Screenshot_125
beach
newBEEES
anthophillia 
newburntest
atmospheric burn test
Screenshot_146
chapel courtroom
Screenshot_142
dodgeball
newemptycourt
empty court
newfetishschool
erp school sim
newfuel
fuel tanks
Screenshot_97
holdout bunker
Screenshot_116
kobayashi maru
Screenshot_144
new lounge 
Screenshot_147
medical sim 
newmedievaltourney
medieval tournament
Screenshot_145
picnic
Screenshot_124
winterwonderland (made by uomo)
Screenshot_107
target practice
newthunderdomesim
thunder dome
newwildlifesim
wildlife sim
winterv3
photobooth

New Station Holodecks:
meta9000
Meta (moved maint to the right, and changed the shape of the airlock, the room below the holodeck is now bigger since it was merged with a hallway)
delta9000
Delta (not many changes, just bigger, and no adjacent carp)
icebox9000
Icebox (moved maint around a bit)
pubby9000
Pubby (lengthened top maint hallway to meet it, and moved bottom maint to the right)

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
idk man, making every other holodeck sim feel like an almost empty room in exchange of the chess sim no one uses to feel realistic?
20 holoeswords and 4 claymores might be a slipshod inkling

**Comment by Kylerace** [comment]
> idk man, making every other holodeck sim feel like an almost empty room in exchange of the chess sim no one uses to feel realistic?
> 20 holoeswords and 4 claymores might be a slipshod inkling
its important to note that ALL of those disappear if anyone changes the holodeck sim, and thats assuming the safety's are off. It takes four hits with a holo esword to down someone with stamina damage, and each hit doesnt slow people down that much. holo eswords in my experience are pretty hard to use for stunning. I dont see a problem with having more holoeswords, since only one round type even has a potential use for the extra swords (assuming all antags used a holosword together, and that no one changes the sim).
However the extra claymores are pretty strong, and i should get rid of two. I assumed that all holo melee weapons only dealt stamina damage.

**Comment by Fikou** [comment]
they all look more ugly empty and dumb

**Comment by Tharcoonvagh** [comment]
Swap the dark clown and mime on the chess sim
Isnt the bee sim supposed to have a beekeper suit and hat?

**Comment by Tharcoonvagh** [comment]
Also yeah they do look kinda empty

**Comment by Kylerace** [comment]
The clown is the king and the mime is the queen, so i cant swap their positions without violating the chess starting position, also the bee sim does have a beekeeping suit, but for some reason it doesn't look like one
beeeeee

**Comment by Kylerace** [comment]
newerlounge
newfiringrange
newwinterthing
Here are some of the improvements im working on to make things less empty, dont really know what to do with the winter one, since its original doesnt have a whole lot of stuff to expand on, also not quite sure what to do with the chapel courtroom

**Comment by ATH1909** [comment]
Nice.

Could you add a second hoop to the bottom of the basketball court simulation, or can hoops only ever face south?

**Comment by Kylerace** [comment]
Unfortunately no, there is no northwards facing hoop that I can find, there are east/west ones but since the sims are 10 tiles tall that would make the hoops off center

**Comment by vkalls** [comment]
Someone really wanted to play real chess huh? 👀

**Comment by ArcaneMusic** [comment]
Did you remove the maint door off of icebox holodeck?
I also can't tell, but are the rechargers still there on the kobiyashi maru? That's one of the few reasons people use that holodeck loadout.
Also, blocking a maint door by default with movables is kinda cringe tbh

**Comment by Kylerace** [comment]
Whoops didnt see those canvasses, thank you. Also a recharger has been re added to kobayashi maru and icebox has a maint door again.
newermetamaint
newerkobayashi
newericebox

**Comment by Tharcoonvagh** [comment]
Wait im dumb aaa

**Comment by trollbreeder** [comment]
if you make it 10x10, you can make space chess and space checkers have a whole ring around, it looks nicer and allows space for 8 more captured pieces

**Comment by Kylerace** [comment]
Originally I did that, and it does look nicer when it comes to the checkers and chess sims, but many of the other sims were designed for there to be an odd number of horizontal tiles. For example, the chapel courtroom has the judge sit in the exact middle, the thunderdome 1218 sim has the crown in the exact middle, kobayashi maru has the front chair in the center, etc. If it was 10 tiles horizontally everything would have to be either to the left or right of the center, so I decided to make it 9x10.

**Comment by GravitronBlue** [comment]
Mind verifying that the skateboards don't become permanant once they're 'deployed'?
Pretty concerned that their holo-quality doesn't carry over when placed.

**Comment by Kylerace** [comment]
Unfortunately I think they do disappear, with both this and the invisible table bug the skate park is unusable. They'll have to be fixed before I can re add it

**Comment by Kylerace** [comment]
Update: Massive thanks to uomu for https://github.com/tgstation/tgstation/pull/52924#issue-467665975 and for fixing the skateboard problem, so now the skating park sim can be re added.

**Comment by Kylerace** [comment]
gameplay of the skate park sim

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/scooter.dm [hunk 1]
```diff
@@ -149,16 +149,19 @@
 
 /obj/vehicle/ridden/scooter/skateboard/MouseDrop(atom/over_object)
 	. = ..()
-	var/mob/living/carbon/M = usr
-	if(!istype(M) || M.incapacitated() || !Adjacent(M))
+	var/mob/living/carbon/Skater = usr
+	if(!istype(Skater) || Skater.incapacitated() || !Adjacent(Skater))
```
[the remark this answers is not in the document]
Comment (Kylerace) [reply code/modules/vehicles/scooter.dm]:
```
/obj/vehicle/ridden/scooter/skateboard/MouseDrop(atom/over_object)
	. = ..()
	var/mob/living/carbon/Skater = usr
	if(!istype(Skater))
		return
	if (over_object == Skater)
		pick_up_board(Skater)

/obj/vehicle/ridden/scooter/skateboard/proc/pick_up_board(mob/living/carbon/Skater)
	if (Skater.incapacitated() || !Adjacent(Skater))
		return
	if(has_buckled_mobs())
		to_chat(Skater, "<span class='warning'>You can't lift this up when somebody's on it.</span>")
		return
	var/board = new board_item_type(get_turf(Skater))
	Skater.put_in_hands(board)
	qdel(src)
```

**Pull request merged**